### PR TITLE
document makefile variables in readme

### PR DESCRIPTION
**Pull request by shubhamkumar13**
The PR helps to document the global variables used in the Makefile while building the dependencies and running the benchmarks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -329,3 +329,42 @@ sed. `sed` will default to a BSD sed on OS X. One way to make things
 work on OS X is to install GNU sed with homebrew and then update the
 `PATH` you run sandmark with to pick up the GNU version.
 
+## Makefile Variables
+
+### Variables
+- BUILD_BENCH_TARGET : benchmark build target type which helps in filtering out PACKAGES - buildbench for serial benchmarks and multibench for multicore benchmarks
+- RUN_CONFIG_JSON : run configuration json file that describes the benchmarks : run_config.json, multicore_parallel_run_config.json, multicore_effects_run_config.json, multicore_parallel_navajo_run_config.json
+- RUN_BENCH_TARGET : add wrapper which are defined inside the RUN_CONFIG_JSON
+- BUILD_ONLY : everything sans building benchmarks
```
Comment (shakthimaan) [inline README.md:338]: It might be good to explicitly specify what it builds, than the other way around.

File: README.md [hunk 2]
```diff
@@ -329,3 +329,42 @@ sed. `sed` will default to a BSD sed on OS X. One way to make things
...
+- ITER : number of iterations for running the benchmark
+- PRE_BENCH_EXEC : pre benchmark wrappers
```
Comment (shakthimaan) [inline README.md:340]: Might be good to give a more explicit description.

File: README.md [hunk 3]
```diff
@@ -329,3 +329,42 @@ sed. `sed` will default to a BSD sed on OS X. One way to make things
...
+- CONTINUE_ON_OPAM_INSTALL_ERROR : allow benchmarks to continue even if the opam package install errors out
+- WRAPPER : get the wrapper out of `run_<wrapper>`
+- PACKAGES : list of all the sandmark dependencies
+- DEPENDENCIES : list of the ubuntu dependencies
```
Comment (shakthimaan) [inline README.md:344]: s/ubuntu/Ubuntu/

File: README.md [hunk 4]
```diff
@@ -329,3 +329,42 @@ sed. `sed` will default to a BSD sed on OS X. One way to make things
...
+- PIP_DEPENDENCIES : list of the python dependencies
+- SYS_DUNE_BASE_DIR : substitute path of dune with the installed dune
```
Comment (shakthimaan) [inline README.md:346]: Path of the system installed dune for use with benchmarking.

File: README.md [hunk 5]
```diff
@@ -329,3 +329,42 @@ sed. `sed` will default to a BSD sed on OS X. One way to make things
...
+- ENVIRONMENT : get the environment variables present in the wrappers section of
```
Comment (shakthimaan) [inline README.md:347]: section of?

File: README.md [hunk 6]
```diff
@@ -329,3 +329,42 @@ sed. `sed` will default to a BSD sed on OS X. One way to make things
...
+- PRE_BENCH_EXEC : pre benchmark wrappers
+- CONTINUE_ON_OPAM_INSTALL_ERROR : allow benchmarks to continue even if the opam package install errors out
+- WRAPPER : get the wrapper out of `run_<wrapper>`
+- PACKAGES : list of all the sandmark dependencies
+- DEPENDENCIES : list of the ubuntu dependencies
+- PIP_DEPENDENCIES : list of the python dependencies
+- SYS_DUNE_BASE_DIR : substitute path of dune with the installed dune
+- ENVIRONMENT : get the environment variables present in the wrappers section of 
+- OCAML_CONFIG_OPTION : looks inside ocaml variant for `configure` parameter
+- OCAML_RUN_PARAM : looks inside ocaml variant for `run_param` parameter
```
Comment (shakthimaan) [inline README.md:349]: The runtime parameters defined in ocaml-variants/.

File: README.md [hunk 7]
```diff
@@ -329,3 +329,42 @@ sed. `sed` will default to a BSD sed on OS X. One way to make things
...
+
+### used in building dependencies
```
Comment (shakthimaan) [inline README.md:351]: Used in building the compiler and its dependencies.

File: README.md (hunk 3 above)
Comment (shubhamkumar13) [reply README.md]: I'm not sure what this means?

File: README.md [hunk 8]
```diff
@@ -329,3 +329,42 @@ sed. `sed` will default to a BSD sed on OS X. One way to make things
...
+- RUN_CONFIG_JSON : run configuration json file that describes the benchmarks : run_config.json, multicore_parallel_run_config.json, multicore_effects_run_config.json, multicore_parallel_navajo_run_config.json
+- RUN_BENCH_TARGET : add wrapper which are defined inside the RUN_CONFIG_JSON
+- BUILD_ONLY : everything sans building benchmarks
+- ITER : number of iterations for running the benchmark
+- PRE_BENCH_EXEC : pre benchmark wrappers
+- CONTINUE_ON_OPAM_INSTALL_ERROR : allow benchmarks to continue even if the opam package install errors out
+- WRAPPER : get the wrapper out of `run_<wrapper>`
+- PACKAGES : list of all the sandmark dependencies
+- DEPENDENCIES : list of the ubuntu dependencies
+- PIP_DEPENDENCIES : list of the python dependencies
```
[the remark this answers is not in the document]
Comment (shubhamkumar13) [reply README.md]: similar to ubuntu comment above I'm not sure what this means

File: README.md (hunk 3 above)
Comment (shakthimaan) [reply README.md]: Use capital letter for nouns. Instead of `ubuntu`, it should be `Ubuntu`. "s///" is a sed expression.

File: README.md [hunk 9]
```diff
@@ -329,3 +329,22 @@ sed. `sed` will default to a BSD sed on OS X. One way to make things
 work on OS X is to install GNU sed with homebrew and then update the
 `PATH` you run sandmark with to pick up the GNU version.
 
+## Makefile Variables
+
+| Name | Description | Default Values | Where in sandmark |
```
Comment (shakthimaan) [inline README.md:334]: Instead of `Where in sandmark`, you can say `Usage`.

File: README.md [hunk 10]
```diff
@@ -329,3 +329,22 @@ sed. `sed` will default to a BSD sed on OS X. One way to make things
...
+| ---- | ----------- | -------------- | ----------------- |
+| BUILD_BENCH_TARGET | Target selection for sequential (buildbench) and parallel (multibench) benchmarks | `buildbench` | building benchmark |
+| RUN_CONFIG_JSON | Input file selection that contains the list of benchmarks | `run_config.json` | executing benchmark |
+| RUN_BENCH_TARGET | The executable to be used to run the benchmarks | `run_orun` | executing benchmark |
+| BUILD_ONLY | if the value is equal to 0 then execute the benchmarks otherwise skip the benchmark execution and exit the sandmark build process | 0 | building benchmark |
```
Comment (shakthimaan) [inline README.md:339]: s/if/If/. Start with a capital letter.

**Pull request merged**